packages/server-core/src/user/accept-invite/accept-invite.service.ts

```diff
@@ -22,23 +23,31 @@ declare module '@etherealengine/common/declarations' {
  * @param next
  * @returns redirect url to the client
  */
-
-function redirect(req, res, next): any {
+async function redirect(ctx, next) {
   try {
-    if (res.data.error) return res.redirect(`${config.client.url}/?error=${res.data.error as string}`)
-    let link = `${config.client.url}/auth/magiclink?type=login&token=${res.data.token as string}`
-    if (res.data.locationName) {
-      let path = `/location/${res.data.locationName}`
-      if (res.data.inviteCode)
-        path += path.indexOf('?') > -1 ? `&inviteCode=${res.data.inviteCode}` : `?inviteCode=${res.data.inviteCode}`
-      if (res.data.spawnPoint)
-        path += path.indexOf('?') > -1 ? `&spawnPoint=${res.data.spawnPoint}` : `?spawnPoint=${res.data.spawnPoint}`
-      if (res.data.spectate)
-        path += path.indexOf('?') > -1 ? `&spectate=${res.data.spectate}` : `?spectate=${res.data.spectate}`
-      if (res.data.instanceId) path += `&instanceId=${res.data.instanceId}`
-      link += `&path=${path}`
+    const { data } = ctx.res
```

Comment:
AcceptInvite.get is not returning data properly, or the data is not on on ctx.res.data. When I accept an invite, whether it runs properly or returns an error, ctx.res.data is undefined, leading the next line to throw an error since it's trying to read a value of undefined.

To test this, make sure that your user is an admin. Got to /admin/invites. Create an invite; it can be any type. Use an email address for the recipient. It won't actually send the email since you don't have SMTP credentials, but if you open up the invite table in the database, there should be be one in there. Go to `https://localhost:3030/a-i/<inviteId>?<invitePasscode>` to attempt to accept the invite.

Reply:
i never tested this manually, i assumed the test was covered in npm run test, ill check it out